libkernelflinger/security.c

```diff
@@ -216,6 +218,123 @@ struct rot_data_t* get_rot_data()
 	return &rot_data;
 }
 
+char* strrpl(CHAR8 *in, const char *src, const char *dst){
+    CHAR8 *p = in;
+    int n = 0;
+    while(*p !='\0'){
```

Comment:
Your replace algo is a little complex. Can do like this:

while(*p != '\0'){
    if(*p == src)
        *p = dst;
    p++;
}